### PR TITLE
add register manage APIs

### PR DESCRIPTION
**Pull request by baixinsui**

### CODE REVIEW

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java [hunk 1]
```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
...
+    }
+
+
+    /**
+     * Get registered service using id.
+     *
+     * @param id id of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:230]: Tag name should be just "Service Vendor" for all vendor related services

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java [hunk 2]
```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
...
+            description = "APIs to get detail the registered services.")
+    @Operation(description = "Get registered service using id.")
+    @GetMapping(value = "/register/service/{id}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public Response detail(@PathVariable("id") String id) {
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:236]: we must return OCL object of the service.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java [hunk 3]
```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
...
-    @Tag(name = "Service Vendor",
-            description = "APIs to publish the managed services.")
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to manage register services.")
+    @Operation(description = "List cloud service providers.")
+    @GetMapping(value = "/register/providers",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @Transactional
+    public Response providers() {
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:75]: we already have the providers as an enum and any client can/must simply use  that directly. I think we dont need this extra service.

File: runtime/src/main/java/org/eclipse/xpanse/runtime/XpanseApplication.java [hunk 4]
```diff
@@ -21,6 +22,7 @@
 @EnableJpaRepositories(basePackages = "org.eclipse.xpanse")
 @EntityScan(basePackages = "org.eclipse.xpanse")
 @ComponentScan(basePackages = "org.eclipse.xpanse")
+@EnableJpaAuditing
```
Comment (swaroopar) [inline runtime/src/main/java/org/eclipse/xpanse/runtime/XpanseApplication.java:25]: do we need this?

File: modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+     * @param registerServiceEntity register service entity
+     */
+    @Override
+    public void remove(RegisterServiceEntity registerServiceEntity) {
+        registerServiceRepository.delete(registerServiceEntity);
+    }
+
+
+    private String getRegisterServicePrimaryKey(RegisterServiceEntity registerService) {
+        StringBuilder stringBuilder = new StringBuilder();
```
Comment (swaroopar) [inline modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java:168]: Primary key is the UUID in the entity. Do we change it to a different one now?

**Review by swaroopar — CHANGES_REQUESTED** [review]
have written some comments after initial review.

File: runtime/src/main/java/org/eclipse/xpanse/runtime/XpanseApplication.java (hunk 4 above)
Comment (baixinsui) [reply runtime/src/main/java/org/eclipse/xpanse/runtime/XpanseApplication.java:25]: It is used  to automatically assign values to database column attributes, such as some creation time and modification time

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java (hunk 3 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:75]: By calling this service. When we change the list of providers,  we only need to modify the enum on the server side, not the client side.

File: modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java (hunk 5 above)
Comment (baixinsui) [reply modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java:168]: This method is used to find the registered service with the same service name, version, service version and service provider in the database when a new service is registering.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java (hunk 2 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:236]: Ok，I will set the OCL object into the Response directly.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java (hunk 1 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:230]: Ok.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java [hunk 6]
```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
...
+    /**
+     * Unregister registered service using id.
+     *
+     * @param id id of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs unregister registered service.")
+    @Operation(description = "Unregister registered service using id.")
+    @DeleteMapping("/register/service/{id}")
```
Comment (iskey) [inline modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:191]:
```suggestion
    @DeleteMapping("/register/{id}")
```
I think /register/{id} is enought.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java [hunk 7]
```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
...
+     * List registered service with query params.
+     *
+     * @param cspName     name of cloud service provider.
+     * @param serviceName name of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to list the managed services.")
+    @Operation(description = "List registered service with query params.")
+    @GetMapping(value = "/register/services",
```
Comment (iskey) [inline modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:216]:
```suggestion
    @GetMapping(value = "/register",
```
Did `/register` works here?

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java [hunk 8]
```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
...
+    /**
+     * Get registered service using id.
+     *
+     * @param id id of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to get detail the registered services.")
+    @Operation(description = "Get registered service using id.")
+    @GetMapping(value = "/register/service/{id}",
```
Comment (iskey) [inline modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:245]:
```suggestion
    @GetMapping(value = "/register/{id}",
```

File: modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java (hunk 5 above)
Comment (baixinsui) [reply modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java:168]: No. This method is just used to find the registered service with the same service name, version, service version and service provider in the database when a new service is registering.  If the same registered service is found, throw exception to show the service already registered.

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/YamlJackson2HttpMessageConverterConfiguration.java [hunk 9]
```diff
@@ -1,35 +1,35 @@
...
-                    MediaType.parseMediaType("application/yaml"),
-                    MediaType.parseMediaType("application/yml"));
-        }
-    }
-
-    @Bean
-    public AbstractJackson2HttpMessageConverter fastJsonHttpMessageConverters() {
-        return new YamlJackson2HttpMessageConverter();
-    }
+/*
```
Comment (iskey) [inline modules/api/src/main/java/org/eclipse/xpanse/api/config/YamlJackson2HttpMessageConverterConfiguration.java:1]: Do we have anything changed here?

File: modules/api/src/main/java/org/eclipse/xpanse/api/exceptions/CommonExceptionHandler.java [hunk 10]
```diff
@@ -60,20 +63,38 @@ public Response handleConstraintViolationException(ConstraintViolationException
...
+        return Response.errorResponse(ResultCode.BAD_PARAMETERS, failMessage);
     }
 
+    /**
+     * Exception handler for Exception.
+     */
     @ExceptionHandler({Exception.class})
     @ResponseStatus(HttpStatus.NOT_FOUND)
     public Response handleException(Exception ex) {
+        ex.printStackTrace();
```
Comment (iskey) [inline modules/api/src/main/java/org/eclipse/xpanse/api/exceptions/CommonExceptionHandler.java:96]: I think we shouldn't keep this, just use a log instead.

**Review by iskey — CHANGES_REQUESTED** [review]

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/YamlJackson2HttpMessageConverterConfiguration.java (hunk 9 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/config/YamlJackson2HttpMessageConverterConfiguration.java:1]: No，I just move it into package config

File: modules/api/src/main/java/org/eclipse/xpanse/api/exceptions/CommonExceptionHandler.java (hunk 10 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/exceptions/CommonExceptionHandler.java:96]: Sure. I will delete it.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java (hunk 8 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:245]: I will delete service in the URL path.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java (hunk 3 above)
Comment (swaroopar) [reply modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:75]: Ok. in this case. the issue is applicable to all enums. Not just this one. In my opinion, the dependent clients must do an update if needed.

File: modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java (hunk 5 above)
Comment (swaroopar) [reply modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java:168]: ok understood. then we can rename the method to something else which does not involve 'primary key'.

File: modules/api/pom.xml [hunk 11]
```diff
@@ -62,5 +62,11 @@
             <artifactId>junit-jupiter-engine</artifactId>
             <scope>test</scope>
         </dependency>
+      <dependency>
+        <groupId>org.eclipse.xpanse.modules</groupId>
+        <artifactId>service</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
```
Comment (swaroopar) [inline modules/api/pom.xml:68]: we can use the project.version instead of fixed version

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added.

File: modules/api/pom.xml (hunk 11 above)
Comment (baixinsui) [reply modules/api/pom.xml:68]: Yes. I will replace the fixed version.

File: modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java (hunk 5 above)
Comment (baixinsui) [reply modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java:168]: Yes. I will change the method name.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java (hunk 3 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java]: Now the UI client shows baseed on return data. I remove the API

**Review by swaroopar — APPROVED** [review]
Thanks. /LGTM

**Review by iskey — APPROVED** [review]
looks good.

**Pull request merged**